### ISSUE TITLE
DriverError ( Could not connect to address ... )

### ISSUE THREAD

**Issue by Xatenev** [blackbeam/rust-mysql-simple#150]
Hello,

I am using the mysql crate in version 14.0.0 on windows 10.

When calling 

`Pool::new("mysql://test:test@127.0.0.1:3306/test").unwrap()`

I get an error:

> thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DriverError { Could not connect to address `\\.\pipe\F:/Dev/xampp/mysql/mysql.sock': The System cannot find the supplied file. (os error 2) }', libcore\result.rs:945:5
> note: Run with `RUST_BACKTRACE=1` for a backtrace.
> thread 'main' panicked at 'Thread pool worker panicked', C:\Users\Megaport\.cargo\registry\src\github.com-1ecc6299db9ec823\scoped_threadpool-0.1.9\src\lib.rs:236:13

When changing the DB name for example to garbage I get

>  ERROR 1049 (42000): Unknown Database ... 

When changing the password to garbage I get

> Access denied for user ...

so it seems to be able to connect just fine. 

Any ideas why this error appears?

Thanks!

PS: I verified via `RUST_BACKTRACE=1 cargo run` that the error is **really** in that specific line.

**Comment by blackbeam** [comment]
Hi.

Socket names on windows usually looks like `mysql`, but in your case it's `F:/Dev/xampp/mysql/mysql.sock` and this looks strange to me. Paths like this are more like unix socket paths so i bet the issue is with mysql server configuration. Could you please execute this query to examine returned string: `SELECT @@socket`? Also could you please dump your my.cnf here?

Or if you just want driver to work, then please change your connection string to this:

```rust
Pool::new("mysql://test:test@127.0.0.1:3306/test?prefer_socket=false").unwrap()
```

**Comment by blackbeam** [comment]
Hmm..
That's strange because according to the docs the default socket value on Windows is `MySql`:

```
--socket=file_name, -S file_name

On Unix, the name of the Unix socket file to use, for connections made using a named pipe to a local server. The default Unix socket file name is /tmp/mysql.sock.

On Windows, the name of the named pipe to use, for connections to a local server. The default Windows pipe name is MySQL. The pipe name is not case-sensitive.

The server must be started with the --enable-named-pipe option to enable named-pipe connections.
```

Or maybe it's because of missing `--enable-named-pipe`..

Anyway. Looks like driver should fallback to TCP in this case, so thanks for report.

**Comment by jens1o** [comment]
suffered from the same issue, also on Windows.

**Comment by 0xpr03** [comment]
I have a similar issue on linux.
I'm port forwarding to a virtual machine for testing but on my system (where I run cargo test) a mariadb instance is already running ( [KDE Akonadi](https://community.kde.org/KDE_PIM/Akonadi) ), so somewhere a configuration exists that prefers a socket.

Using
```rust
builder.ip_or_hostname(Some("foo"))
            .db_name(Some(TEST_DB))
            .user(Some(user))
            .pass(password)
            .tcp_port(3333)
            .ip_or_hostname(Some("127.0.0.1"));
```
without `.prefer_socket(false)` will fail with
```log
Could not connect to address \`/var/run/mysqld/mysqld.sock': No such file or directory (os error 2) }', libcore/result.rs:945:5`
```

Based on [this answer](https://serverfault.com/a/337928) I would expect this to work without `prefer_socket(false)` when using `127.0.0.1`

**Comment by blackbeam** [comment]
Fixed in v14.1.0

**Comment by jens1o** [comment]
https://github.com/blackbeam/rust-mysql-simple/commit/47cc68384110388ed751299b7c9a075c5056056a
